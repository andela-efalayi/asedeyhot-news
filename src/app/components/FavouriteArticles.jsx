import React from 'react';
import PropTypes from 'prop-types';
import { Card, CardMedia, CardText } from 'material-ui/Card';

/**
 * Displays all favourite articles on Favorites page
 * @function ArchiveChard
 * @param {object} articles
 * @return {object} react-component
 */

const FavouriteArticles = ({ articles }) => {
  if (articles === null) {
    return (
      <div>
        <h4>You have no saved articles.</h4>
      </div>
    );
  }
  return (
    <div className="row">
      {Object.entries(articles).map(([key, value]) => (
        <div className="col-xs-6 col-sm-6 col-md-3" key={key}>
          <Card className="card">
            <CardMedia>
              <img src={value.urlToImage} alt=""/>
            </CardMedia>
            <CardText>
              <h6>{value.title}
                <span>...<a href={value.url}>more</a></span>
              </h6>
              <p>Author: {value.author}</p>
            </CardText>
          </Card>
      </div>
    ))}
  </div>
  );
};

FavouriteArticles.propTypes = {
  articles: PropTypes.object
};

export default FavouriteArticles;
