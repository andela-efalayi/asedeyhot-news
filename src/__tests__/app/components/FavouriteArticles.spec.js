import React from 'react';
import { shallow, mount } from 'enzyme';
import FavouriteArticles from '../../../app/components/FavouriteArticles.jsx';

let emptyResponseFromFirebase;

const favouriteArticles = {
  title: {
    author: 'BBC News',
    description: 'The party says it would prioritise',
    url: 'http://www.bbc.co.uk/news/uk-politics-39698465'
  }
};

describe('FavouriteArticles.jsx', () => {
  it('should match component snapshot',
    () => {
      const wrapper = shallow(
          <FavouriteArticles articles={favouriteArticles}/>
      );
      expect(wrapper).toMatchSnapshot();
    });
  it('should contain "You have no saved articles" for an empty response',
    () => {
      const wrapper = mount(
          <FavouriteArticles articles={emptyResponseFromFirebase}/>
      );
      expect(wrapper.containsMatchingElement(
        <p>You have no saved articles.</p>)
      ).toEqual(true);
    });
});
