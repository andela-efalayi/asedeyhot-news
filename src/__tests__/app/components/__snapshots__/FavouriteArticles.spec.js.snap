// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavouriteArticles.jsx should match component snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="row"
>
    <div
        className="col-xs-6 col-sm-6 col-md-3"
    >
        <Card
            className="card"
            expandable={false}
            expanded={null}
            initiallyExpanded={false}
        >
            <CardMedia>
                <img
                    alt=""
                    src={undefined}
                />
            </CardMedia>
            <CardText>
                <h6>
                    <span>
                        ...
                        <a
                            href="http://www.bbc.co.uk/news/uk-politics-39698465"
                        >
                            more
                        </a>
                    </span>
                </h6>
                <p>
                    Author: 
                    BBC News
                </p>
            </CardText>
        </Card>
    </div>
</div>,
  "nodes": Array [
    <div
      className="row"
>
      <div
            className="col-xs-6 col-sm-6 col-md-3"
      >
            <Card
                  className="card"
                  expandable={false}
                  expanded={null}
                  initiallyExpanded={false}
            >
                  <CardMedia>
                        <img
                              alt=""
                              src={undefined}
                        />
                  </CardMedia>
                  <CardText>
                        <h6>
                              <span>
                                    ...
                                    <a
                                          href="http://www.bbc.co.uk/news/uk-politics-39698465"
                                    >
                                          more
                                    </a>
                              </span>
                        </h6>
                        <p>
                              Author: 
                              BBC News
                        </p>
                  </CardText>
            </Card>
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <FavouriteArticles
        articles={
                Object {
                        "title": Object {
                          "author": "BBC News",
                          "description": "The party says it would prioritise",
                          "url": "http://www.bbc.co.uk/news/uk-politics-39698465",
                        },
                      }
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "articles": Object {
            "title": Object {
              "author": "BBC News",
              "description": "The party says it would prioritise",
              "url": "http://www.bbc.co.uk/news/uk-politics-39698465",
            },
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="row"
>
          <div
                    className="col-xs-6 col-sm-6 col-md-3"
          >
                    <Card
                              className="card"
                              expandable={false}
                              expanded={null}
                              initiallyExpanded={false}
                    >
                              <CardMedia>
                                        <img
                                                  alt=""
                                                  src={undefined}
                                        />
                              </CardMedia>
                              <CardText>
                                        <h6>
                                                  <span>
                                                            ...
                                                            <a
                                                                      href="http://www.bbc.co.uk/news/uk-politics-39698465"
                                                            >
                                                                      more
                                                            </a>
                                                  </span>
                                        </h6>
                                        <p>
                                                  Author: 
                                                  BBC News
                                        </p>
                              </CardText>
                    </Card>
          </div>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="row"
>
          <div
                    className="col-xs-6 col-sm-6 col-md-3"
          >
                    <Card
                              className="card"
                              expandable={false}
                              expanded={null}
                              initiallyExpanded={false}
                    >
                              <CardMedia>
                                        <img
                                                  alt=""
                                                  src={undefined}
                                        />
                              </CardMedia>
                              <CardText>
                                        <h6>
                                                  <span>
                                                            ...
                                                            <a
                                                                      href="http://www.bbc.co.uk/news/uk-politics-39698465"
                                                            >
                                                                      more
                                                            </a>
                                                  </span>
                                        </h6>
                                        <p>
                                                  Author: 
                                                  BBC News
                                        </p>
                              </CardText>
                    </Card>
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <FavouriteArticles
    articles={
        Object {
            "title": Object {
              "author": "BBC News",
              "description": "The party says it would prioritise",
              "url": "http://www.bbc.co.uk/news/uk-politics-39698465",
            },
          }
    }
/>,
}
`;
